drivers/include/sdmmc/sdmmc.h

```diff
@@ -0,0 +1,1905 @@
+ *   indicates that the SDIO/SD/MMC peripheral supports the Auto-CMD12
+ *   feature, i.e. CMD12 is sent automatically to stop the transmission in
+ *   multiple block operations. This feature shall be provided by the MCU
+ *   if supported.
+ *
+ * - <b>`periph_sdmmc_hs`</b> (`HAS_PERIPH_SDMMC_HS` in Kconfig)
+ *   indicates that the SDIO/SD/MMC peripheral supports MMC/eMMCs. This
+ *   feature shall be provided by the MCU if supported.
+ *
+ * - <b>`periph_sdmmc_mmc`</b> (`HAS_PERIPH_SDMMC_MMC` in Kconfig)
+ *   indicates that the SDIO/SD/MMC peripheral supports the high speed
+ *   access, that is 50 MHz for SD and 52 MHz for MMC. This feature shall be
+ *   provided by the MCU if supported.
+ *
+ * Some functionalities of the SDIO/SD/MMC Device API must be explicitly
+ * enabled via modules:
+ *
+ * - <b>`sdmmc_mmc`</b> (`MODULE_SDMMC_MMC` in Kconfig)
+ *   enables the support for MMCs/eMMCs.
+ *
+ * - <b>`periph_sdmmc_8bit`</b> (`MODULE_PERIPH_SDMMC_8BIT` in Kconfig)
+ *   enables the 8-bit bus width support. It requires the corresponding
+ *   of the board.
+ *
+ * @{
+ *
+ * @file
+ * @brief       SDIO/SD/MMC device API using a low-level peripheral driver
+ *
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ */
+
+#ifndef SDMMC_SDMMC_H
+#define SDMMC_SDMMC_H
+
+#include <errno.h>
+
+#include "assert.h"
+#include "macros/units.h"
+#include "periph_conf.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   CPU-specific requirements for SDIO/SD/MMC buffers
+ *
+ * Can be overridden by periph_cpu if needed by the SDMMC/SDIO peripheral.
+ * Example usage:
+ * ```
+ * SDMMC_CPU_DMA_REQUIREMENTS uint8_t buffer[64];
+ * ```
+ */
+#ifndef SDMMC_CPU_DMA_REQUIREMENTS
+#define SDMMC_CPU_DMA_REQUIREMENTS
+#endif
+
+/**
+ * @brief   Instantiation type for SDIO/SD/MMC buffers
+ *
+ * Example usage:
+ * ```
+ * sdmmc_buf_t buffer[SDMMC_SDHC_BLOCK_SIZE];
+ * ```
+ */
+#define sdmmc_buf_t     SDMMC_CPU_DMA_REQUIREMENTS uint8_t
+
+/**
+ * @brief   Size of a single data block on SDHC/SDXC Cards in bytes
+ *
+ * The size of a data block depend on type and mode of the card. SDSC/SDXC
+ * use a fix size of 512 bytes.
+ */
+#define SDMMC_SDHC_BLOCK_SIZE   (512)
+
+/**
+ * @name    Command and Response related definitions
+ * @{
+ */
+
+/**
+ * @brief   Command index
+ */
+#define SDMMC_CMD(n)        (n)
+
+/**
+ * @brief   Application speecific command index prefix (Bit is 7 used)
+ */
+#define SDMMC_ACMD_PREFIX   (1 << 7)
+
+/**
+ * @brief   Application specific command index
+ */
+#define SDMMC_ACMD(n)       (SDMMC_ACMD_PREFIX | SDMMC_CMD(n))
+
+/**
+ * @brief   SDIO/SD/MMC Commands
+ *
+ * Commands used by the driver. Command indices are in range [0..63].
+ * Application specific command indices are ORed with
+ * @ref SDMMC_ACMD_PREFIX (0x80) and thus marked as application specific.
+ * @ref SDMMC_ACMD_PREFIX can be used to check wther a given command index
+ * specifies an application specific command.
+ */
+enum {
```

Comment:
Hm, not sure because the enum doesn't have all commands defined and the application might want to send its own commands :thinking: On the other hand the `enum` type should be compatible with `uint8_t` so that the user should still have the option to send its own command indeces.